src/support.cpp

```diff
@@ -19,6 +19,25 @@
 #include "settings/types/AngleRadians.h" //To compute overhang distance from the angle.
 #include "utils/math.h"
 
+namespace
+{
+
+/*
+ * If support_wall_count > 0, then the actual outermost contour of the printed support polygons will be the outer inset of support infill part, offset from the outline on a half of line width.
+ * This function returns the actual offset value.
+ */
+cura::coord_t getActualSupportOffset()
+{
+    const cura::Settings& mesh_group_settings = cura::Application::getInstance().current_slice->scene.current_mesh_group->settings;
+    const cura::ExtruderTrain& infill_extruder = mesh_group_settings.get<cura::ExtruderTrain&>("support_infill_extruder_nr");
+    const cura::coord_t support_line_width = infill_extruder.settings.get<cura::coord_t>("support_line_width");
+    const size_t wall_line_count = infill_extruder.settings.get<size_t>("support_wall_count");
+    const cura::coord_t support_offset = wall_line_count > 0 ? -support_line_width / 2 : 0;
+    return support_offset;
+}
+
+}
```

Comment:
I don't like global functions in the global namespace. I'd move this one in the AreaSupport class...

Reply:
Thank you for review, I will have a look and make changes

Reply:
Was moved to the AreaSupport scope